### PR TITLE
Psalm fixes

### PR DESCRIPTION
**Pull request by orklah**
Following #89, this PR:
- increases Psalm strictness (less "info" rules, "totallyTyped"=true)
- fix issues
- update baseline
- increase type coverage to 100%

Status after this PR:
- Two issues left in baseline
- Two rules left in "info": structural choices. Potentially hard to fix
- Two issues left in CI to be discussed. I believe this is a legitimate issue. I'll flag it on review

### CODE REVIEW

File: src/Token/Stream.php [hunk 1]
```diff
@@ -64,36 +64,44 @@ class PHP_Token_Stream implements ArrayAccess, Countable, SeekableIterator
...
     /**
-     * @var array
+     * @var array|null
      */
     protected $traits;
 
     /**
      * @var array
+     *
+     * @psalm-var array<int, string>
```
Comment (orklah) [inline src/Token/Stream.php]: lineToFunctionMap is set at line 629 with int key and string value but is accessed on line 297 with string key. I think getFunctionForLine should take an int param.

**Comment by sebastianbergmann** [comment]
Thank you for your contribution. I appreciate the time you invested in preparing this pull request. However, I have decided not to merge it as I cannot all changes proposed. Please send individual pull requests, one for each issue identified by Psalm. Thanks!

**Comment by orklah** [comment]
Any indication to which change you don't want? It might save me some time :)

**Comment by sebastianbergmann** [comment]
Eventually I want all of them but I cannot merge all of them right now (because of API changes).

**Pull request closed without merging**